libraries/aws_securityhub_finding.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsSecurityHubFinding < AwsResourceBase
+  name "aws_securityhub_finding"
+  desc 'Verifies settings for an AWS SecurityHub Finding'
+  example "
+  describe aws_securityhub_finding(filters: {title: 'finding title') do
+    it { should exist }
+  end
+  "
+  
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters(required: [:filters])
+    @filters = {filters: opts[:filters]}
+    
+    # Ensure the response will have a single finding.
+    @filters[:max_results] = 1
+    
+    catch_aws_errors do
+      begin
+        resp = @aws.securityhub_client.get_findings **@filters
+        @finding = resp.findings.first.to_h
+        if @finding.empty?
+          raise Aws::SecurityHub::Errors::FindingNotFound.new(resp, "#{@__resource_name__} not found.")
```

Comment:
all [security hub errors](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/SecurityHub/Errors.html) inherit from `ServiceErrors` and `catch_aws_errors` can handle all AWS service errors, so need to have this check